megamek/src/megamek/client/RandomNameGenerator.java

```diff
@@ -356,6 +348,59 @@ public String generate(boolean isFemale) {
         return "Unnamed";
     }
 
+    /**
+     * Generate a single random name split between a given name and surname
+     *
+     * @return - a String[] containing the name,
+     *              with the given name at String[0]
+     *              and the surname at String[1]
+     */
+    public String[] generateGivenNameSurnameSplit() {
+        return generateGivenNameSurnameSplit(isFemale());
+    }
+
+    public String[] generateGivenNameSurnameSplit(boolean isFemale) {
+        String[] name = {"", ""};
+        if ((chosenFaction != null) && (factionLast != null)
+                && (factionFirst != null) && (firstM != null)
+                && (firstF != null) && (last != null)) {
+            // this is a total hack, but for now lets assume that
+            // if the chosenFaction name contains the word "clan"
+            // we should only spit out first names
+            boolean isClan = chosenFaction.toLowerCase().contains("clan");
+
+            Vector<String> ethnicities = factionLast.get(chosenFaction);
+            if ((null != ethnicities) && (ethnicities.size() > 0)) {
+                String eLast = ethnicities.get(Compute.randomInt(ethnicities.size()));
+                // ok now we need to decide on a first name list
+                ethnicities = factionFirst.get(chosenFaction).get(eLast);
+                if ((null != ethnicities) && (ethnicities.size() > 0)) {
+                    String eFirst = ethnicities.get(Compute.randomInt(ethnicities.size()));
+                    // ok now we can get the first and last name vectors
+                    if (isClan) {
+                        eFirst = eLast;
+                    }
+                    Vector<String> firstNames = firstM.get(eFirst);
+                    if (isFemale) {
+                        firstNames = firstF.get(eFirst);
+                    }
```

Comment:
Candidate for ternary operator simplification.

Reply:
I agree, fixed